cores/arduino/HardwareTimer.h

```diff
@@ -0,0 +1,154 @@
+  SOFTWARE.
+
+  Copyright (c) 2019 STMicroelectronics
+  Modified to support Arduino_Core_STM32
+*/
+
+/* Define to prevent recursive inclusion -------------------------------------*/
+#ifndef HARDWARETIMER_H_
+#define HARDWARETIMER_H_
+
+/* Includes ------------------------------------------------------------------*/
+#include "timer.h"
+
+#ifdef HAL_TIM_MODULE_ENABLED
+
+#define  TIMER_CHANNELS 4    // channel5 and channel 6 are not considered here has they don't have gpio output and they don't have interrupt
+
+typedef enum {
+  TIMER_DISABLED,
+  // Output Compare
+  TIMER_OUTPUT_COMPARE,                   // == TIM_OCMODE_TIMING           no output, useful for only-interrupt
+  TIMER_OUTPUT_COMPARE_ACTIVE,            // == TIM_OCMODE_ACTIVE           pin is set high when counter == channel compare
+  TIMER_OUTPUT_COMPARE_INACTIVE,          // == TIM_OCMODE_INACTIVE         pin is set low when counter == channel compare
+  TIMER_OUTPUT_COMPARE_TOGGLE,            // == TIM_OCMODE_TOGGLE           pin toggles when counter == channel compare
+  TIMER_OUTPUT_COMPARE_PWM1,              // == TIM_OCMODE_PWM1             pin high when counter < channel compare, low otherwise
+  TIMER_OUTPUT_COMPARE_PWM2,              // == TIM_OCMODE_PWM2             pin low when counter < channel compare, high otherwise
+  TIMER_OUTPUT_COMPARE_FORCED_ACTIVE,     // == TIM_OCMODE_FORCED_ACTIVE    pin always high
+  TIMER_OUTPUT_COMPARE_FORCED_INACTIVE,   // == TIM_OCMODE_FORCED_INACTIVE  pin always low
+
+  //Input capture
+  TIMER_INPUT_CAPTURE_RISING,             // == TIM_INPUTCHANNELPOLARITY_RISING
+  TIMER_INPUT_CAPTURE_FALLING,            // == TIM_INPUTCHANNELPOLARITY_FALLING
+  TIMER_INPUT_CAPTURE_BOTHEDGE,           // == TIM_INPUTCHANNELPOLARITY_BOTHEDGE
+
+  // Used 2 channels for a single pin. One channel in TIM_INPUTCHANNELPOLARITY_RISING another channel in TIM_INPUTCHANNELPOLARITY_FALLING.
+  // Channels must be used by pair: CH1 with CH2, or CH3 with CH4
+  // This mode is very useful for Frequency and Dutycycle measurement
+  TIMER_INPUT_FREQ_DUTY_MEASUREMENT,
+
+  TIMER_NOT_USED = 0xFFFF  // This must be the last item of this enum
+} TimerModes_t;
+
+typedef enum {
+  TICK_FORMAT, // default
+  MICROSEC_FORMAT,
+  HERTZ_FORMAT,
+} TimerFormat_t;
+
+typedef enum {
+  TICK_COMPARE_FORMAT, // default
+  MICROSEC_COMPARE_FORMAT,
+  HERTZ_COMPARE_FORMAT,
+  PERCENT_COMPARE_FORMAT,  // used for Dutycycle
+  RESOLUTION_8B_COMPARE_FORMAT,  // used for Dutycycle: [0.. 255]
+  RESOLUTION_12B_COMPARE_FORMAT  // used for Dutycycle: [0.. 4095]
+} TimerCompareFormat_t;
+
+// This structure is used to be able to get HardwareTimer instance (C++ class)
+// from handler (C structure) specially for interrupt management
+typedef struct  {
+  void    *__this;  // Those 2 first field must remain in this order at the beginning of the structure
+  TIM_HandleTypeDef handle; // Those 2 first field must remain in this order at the beginning of the structure
```

Comment:
```suggestion
  TIM_HandleTypeDef handle;
```

Reply:
Done

Reply:
Not done

Reply:
Oups, sorry, modification has been removed when I revert commit to change name of '__this' which is kept after all.
Done